test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -395,6 +395,43 @@ public void DuplicateInstrumentRegistration_NoViews_DuplicateInstruments_Differe
             Assert.Equal(20D, metricPoint2.GetHistogramSum());
         }
 
+        [Fact]
+        public void DuplicateInstrumentRegistration_WithViews_DuplicateInstruments_DifferentDescription()
+        {
+            var exportedItems = new List<Metric>();
+
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}");
+
+            // Ensure that the two instruments have different MetricStreamIdentity but the same MetricStreamName
+            var meterProviderBuilder = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddView("duplicateInstrumentName", new MetricStreamConfiguration { Name = "instrumentName", Description = "duplicateInstrument" })
+                .AddInMemoryExporter(exportedItems);
+
+            using var meterProvider = meterProviderBuilder.Build();
+
+            var instrument = meter.CreateCounter<long>("instrumentName");
+            var duplicateInstrument = meter.CreateCounter<long>("duplicateInstrumentName");
+
+            instrument.Add(10);
+            duplicateInstrument.Add(20);
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+            Assert.Single(exportedItems);
```

Comment:
I would expect two items. They should have conflicting names.

See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view

> For each View, if the Instrument could match the instrument selection criteria:
Try to apply the View configuration. If applying the View results in [conflicting metric identities](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#opentelemetry-protocol-data-model-producer-recommendations) the implementation SHOULD apply the View and emit a warning.

Reply:
Though _technically_ there's this annoying statement [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#opentelemetry-protocol-data-model-producer-recommendations)

> If the potential conflict involves a non-identifying property (i.e., description), the producer SHOULD choose the longer string.

We do not do this today.

Reply:
The areas of the spec concerning views and instrument identity still have some rough edges.  The statement above declares description a "non-identifying property". This seems to contradict [this part of the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument) that states:

> Instruments are identified by all of these fields.

That is name, kind, unit, _and_ description are identifying.

My head hurts now. I hope you have a good weekend.